tests/framework/e2e/etcd_process.go

```diff
@@ -62,6 +62,7 @@ type EtcdProcess interface {
 
 type LogsExpect interface {
 	ExpectWithContext(context.Context, string) (string, error)
+	ExpectFunc(ctx context.Context, f func(string) bool) (string, error)
```

Comment:
Why add `ExpectFunc`?

Reply:
Because once I expected multiple strings(different branches may have different logs). But after adding the logs it is not needed. However I think it provides more flexibility and is something good to have so I didn't delete it.

Reply:
It makes no sense to add something which is not used at all.

Please remove it in this PR, and If you really think it's useful, let's discuss it separately.